solana/integration-lib/src/state.rs

```diff
@@ -63,6 +66,41 @@ pub fn get_gatekeeper_address_with_seed(authority: &Pubkey, network: &Pubkey) ->
     )
 }
 
+fn verify_gatekeeper_length(gatekeeper_account_info: &AccountInfo) -> Result<(), GatewayError> {
+    // Length must be same as `GATEKEEPER_ACCOUNT_LENGTH` and have at least one non-zero byte.
+    // Must have one non-zero as being assigned an account with the proper length requires all bytes be zero
+    // Pubkey guarantees one non-zero byte with proper data
+    if gatekeeper_account_info.data_len() != GATEKEEPER_ACCOUNT_LENGTH {
+        return Err(GatewayError::InvalidGatekeeperAccount);
+    }
+
+    Ok(())
+}
+/// Verifies that the gatekeeper account matches the passed in gatekeeper and gatekeeper network
+/// NOTE: This does not check that the gatekeeper is a signer of the transaction.
+pub fn verify_gatekeeper(
+    gatekeeper_account_info: &AccountInfo,
+    gatekeeper: Pubkey,
+    gatekeeper_network: Pubkey,
```

Comment:
Change to take `&Pubkey`s, it's more efficient on compute cycles

Reply:
done